util/election2/etcd/election.go

```diff
@@ -89,6 +98,10 @@ func (e *Election) WithMastership(ctx context.Context) (context.Context, error)
 // Resign releases mastership for this instance. The instance can be elected
 // again using Await. Idempotent, might be useful to retry if fails.
 func (e *Election) Resign(ctx context.Context) error {
+	// Trigger Observe callers to see the update, and cancel mastership contexts.
```

Comment:
etcd.Resign returns `nil` when it is not master. The introduction of a proclaim step which will fail when not master changes that API contract. Does the comment above that it "might be useful to retry if it fails" still hold?

Reply:
Good point. I think that we should either a) return some sort of `election2.NotMaster` error, or b) stick to the contract and return nil if `Proclaim` returns `concurrency.ErrElectionNotLeader`. I am leaning towards b), WDYT?

Reply:
I lean towards B as well

Reply:
Done.